### DISCUSSION TITLE
Urgent Help Needed: Docker API Service Error Related to jaxlib and AVX Instructions

### DISCUSSION THREAD

**Discussion by yesyyesy** [langgenius/dify#7345] [General]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Content

Hi everyone,

I’m facing a critical issue while running a Docker API service on my server. The service is throwing the following error:

“Error: This version of jaxlib was built using AVX instructions, which your CPU and/or operating system do not support. You may be able to work around this issue by building jaxlib from source.”

I’m not very familiar with these technical details and could really use some assistance. Does anyone have experience with this error and know how to resolve it? I’m open to any suggestions or guidance on how to build jaxlib from source or any other potential solutions.
Your help is greatly appreciated. Thank you in advance!

**Comment by thierry-marchisio** [comment]
Hello,

It seems your computer hardware does not support Intel AVX instructions. AVX is quite common now-a-days on Intel CPU.
Using the default jaxlib librairies without aving an AVX compliant Intel processor raises an error because your CPU does not understand what to do with AVX instructions which are Floating Points Numbers Computing Acceleration Features)

Since 2021, AVX instruction has been considered to be the default by jaxlib teams. Since then, the jaxlib library assumes everyone using it has an AVX compliant processor.

*jaxlib wheels are now built to require AVX instructions on x86-64 machines by default. If you want to use JAX on a machine that doesn’t support AVX, you can build a jaxlib from source using the --target_cpu_features flag to build.py. --target_cpu_features also replaces --enable_march_native.*
Source : [https://jax.readthedocs.io/en/latest/changelog.html#jaxlib-0-1-62-march-9-2021](https://jax.readthedocs.io/en/latest/changelog.html#jaxlib-0-1-62-march-9-2021)


To be able to use jaxlib, you must build a your own version of jaxlib that does not use AVX instructions.

These are the standard commands you should execute on a terminal to prepare for the build :
```
git clone https://github.com/google/jax
cd jax
sudo apt install g++ python python3-dev
```
This is the modified build command you should use to build *jaxlib without AVX features support* :

```
python build/build.py --requirements_update --target_cpu_features
```

For more details, or if your build fails due to other specific requirements by your hardware or software installation, please refer to this page : [https://jax.readthedocs.io/en/latest/developer.html#building-from-source](https://jax.readthedocs.io/en/latest/developer.html#building-from-source)

I hope this works on your hardware !
